File: code/ruby/app/controllers/draw_five_controller.rb
class DrawFiveController < ApplicationController
  def index

  end

  def draw
    game = DrawFiveGame.new
    draw_result = game.draw

    @cards = ''
    draw_result.cards.each {|card|
      if @cards.length > 0
        @cards += ', '
      end
      @cards += card.description
    }

    @score = draw_result.score
    @name = ''
  end

  def highscores
    game = DrawFiveGame.new
    @high_scores = game.get_high_scores
  end

  def savescore
    game = DrawFiveGame.new
    game.save_score(params[:name], params[:score])
    redirect_to url_for :action => 'highscores'
  end
end

File: code/c-sharp/x64/src/OopSolidSample/Controllers/DrawFiveController.cs
using System.Web.Mvc;
using OopSolidSample.Domain;
using OopSolidSample.Models.DrawFive;
using MvcContrib.Attributes;
using MvcContrib;

namespace OopSolidSample.Controllers
{
    public class DrawFiveController : Controller
    {
        [AcceptGet]
        public ActionResult Index()
        {
            var model = new DrawFiveDrawViewModel();

            return View();
        }

        [AcceptPost]
        public ActionResult Draw()
        {
            var model = new DrawFiveDrawViewModel();
            var game = new DrawFiveGame();
            var drawResult = game.Draw();
            
            foreach (var card in drawResult.Cards)
            {
                if (!string.IsNullOrEmpty(model.Cards))
                    model.Cards += ", ";
                model.Cards += card.Description;
            }

            model.Score = drawResult.Score;
            return View(model);
        }

        public ActionResult HighScores()
        {
            var game = new DrawFiveGame();
            var model = new DrawFiveHighScoresViewModel();
            model.HighScores = game.GetHighScores();
            return View(model);
        }

        [AcceptPost]
        public ActionResult SaveScore(DrawFiveSaveScoreUpdateModel model)
        {
            var game = new DrawFiveGame();
            game.SaveScore(model.Name, model.Score);
            return this.RedirectToAction(c => c.HighScores());
        }
    }
}